MacOS build fails for an error in SDL2

Hello,

I had an hard time compiling dablin on MacOS Monterey, 12.0.1 (21A559). I followed the instructions in the README, installing the dependencies with Homebrew, but I kept getting this error:
```
$ make
[  2%] Building C object fec/CMakeFiles/fec.dir/encode_rs_char.c.o
[  4%] Building C object fec/CMakeFiles/fec.dir/decode_rs_char.c.o
[  7%] Building C object fec/CMakeFiles/fec.dir/init_rs_char.c.o
[  9%] Linking C static library libfec.a
[  9%] Built target fec
[ 12%] Building C object fec/test/CMakeFiles/rstest.dir/rstest.c.o
[ 14%] Linking C executable rstest
[ 14%] Built target rstest
[ 17%] Building C object fec/test/CMakeFiles/rs_speedtest.dir/rs_speedtest.c.o
[ 19%] Linking C executable rs_speedtest
[ 19%] Built target rs_speedtest
[ 21%] Building CXX object src/CMakeFiles/dablin_gtk.dir/sdl_output.cpp.o
In file included from /Users/alerinaldi/dablin/src/sdl_output.cpp:19:
In file included from /Users/alerinaldi/dablin/src/sdl_output.h:29:
In file included from /usr/local/include/SDL2/SDL.h:32:
In file included from /usr/local/include/SDL2/SDL_main.h:25:
/usr/local/include/SDL2/SDL_stdinc.h:476:5: error: use of undeclared identifier 'memset_pattern4'
    memset_pattern4(dst, &val, dwords * 4);
    ^
1 error generated.
make[2]: *** [src/CMakeFiles/dablin_gtk.dir/sdl_output.cpp.o] Error 1
make[1]: *** [src/CMakeFiles/dablin_gtk.dir/all] Error 2
make: *** [all] Error 2
```

Since this seems more a SDL2 issue, I searched for similar issues but I found only https://github.com/dingusdev/dingusppc/issues/15 that seems to be no longer applying.

Eventually, I edited the [/usr/local/include/SDL2/SDL_stdinc.h](https://fossies.org/linux/SDL2/include/SDL_stdinc.h) removing lines 464 and 465 and replacing line 466 (now 464) with:
```
#if defined(__GNUC__) && defined(__i386__)
```

Doing this, it built without any other issue and it actually works, but I really believe this is not the most correct approach :)

--- a/src/dablin.h
+++ b/src/dablin.h
@@ -19,11 +19,13 @@
 #ifndef DABLIN_H_
 #define DABLIN_H_
 
+// eti_player.h, that indirectly includes SDL.h, must be included before <string> or it won't compile on OS X
+#include "eti_player.h"
+
 #include <signal.h>
 #include <string>
 
 #include "eti_source.h"
-#include "eti_player.h"
 #include "edi_source.h"
 #include "edi_player.h"
 #include "fic_decoder.h"

--- a/src/dablin_gtk.h
+++ b/src/dablin_gtk.h
@@ -19,6 +19,9 @@
 #ifndef DABLIN_GTK_H_
 #define DABLIN_GTK_H_
 
+// eti_player.h, that indirectly includes SDL.h, must be included before <string> or it won't compile on OS X
+#include "eti_player.h"
+
 #include <algorithm>
 #include <list>
 #include <mutex>
@@ -32,7 +35,6 @@
 
 #include "dablin_gtk_sls.h"
 #include "eti_source.h"
-#include "eti_player.h"
 #include "edi_source.h"
 #include "edi_player.h"
 #include "fic_decoder.h"

--- a/src/ensemble_player.h
+++ b/src/ensemble_player.h
@@ -19,6 +19,11 @@
 #ifndef ENSEMBLE_PLAYER_H_
 #define ENSEMBLE_PLAYER_H_
 
+// sdl_output.h, that includes SDL.h, must be included before <string> or it won't compile on OS X
+#ifndef DABLIN_DISABLE_SDL
+#include "sdl_output.h"
+#endif
+
 #include <stdio.h>
 #include <stdint.h>
 #include <mutex>
@@ -30,10 +35,6 @@
 #include "dabplus_decoder.h"
 #include "pcm_output.h"
 #include "tools.h"
-
-#ifndef DABLIN_DISABLE_SDL
-#include "sdl_output.h"
-#endif
 
 
 // --- EnsemblePlayerObserver -----------------------------------------------------------------

--- a/src/sdl_output.h
+++ b/src/sdl_output.h
@@ -19,14 +19,15 @@
 #ifndef SDL_OUTPUT_H_
 #define SDL_OUTPUT_H_
 
+// SDL.h must be included before <string> or it won't compile on OS X
+#include "SDL.h"
+
 #include <stdexcept>
 #include <stdint.h>
 #include <string>
 #include <mutex>
 #include <vector>
 #include <atomic>
-
-#include "SDL.h"
 
 #include "audio_output.h"
 #include "tools.h"

